nncf/data/dataloader.py

```diff
@@ -0,0 +1,103 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Iterable
+from typing import Iterator
+from typing import Callable
+from abc import ABC
+from abc import abstractmethod
+
+from nncf.data.types import DataItem
+from nncf.data.types import ModelInput
+from nncf.data.types import DataSource
+
+
+class DataLoader(ABC):
+    """
+    Describes the interface of the data source that is used by
+    compression algorithms.
+
+    The `DataLoader` object contains the dataset and information
+    about how to transform the data item returned per iteration to
+    the model's expected input.
+    """
+
+    @property
+    @abstractmethod
+    def batch_size(self) -> int:
```

Comment:
I see the batch is coming through the whole code. But I can't see how it is used. @andrey-churkin, can you please explain?